### PR TITLE
CaffeOnSpark Python API

### PR DESCRIPTION
**Pull request by mriduljain**

### CODE REVIEW

File: data/lenet_dataframe_solver.prototxt [hunk 1]
```diff
@@ -20,4 +20,4 @@ max_iter: 10000
 snapshot: 5000
 snapshot_prefix: "mnist_lenet"
 # solver mode: CPU or GPU
-solver_mode: GPU
+solver_mode: CPU
```
Comment (anfeng) [inline data/lenet_dataframe_solver.prototxt]: Python API should not need to change this setting

File: data/lenet_dataframe_train_test.prototxt [hunk 2]
```diff
@@ -9,7 +9,7 @@ layer {
   }
   source_class: "com.yahoo.ml.caffe.ImageDataFrame"
   memory_data_param {
-    source: "hdfs:///projects/machine_learning/image_dataset/mnist_train_dataframe/"
+    source: "/Users/mridul/bigml/mnist_train_dataframe/"
```
Comment (anfeng) [inline data/lenet_dataframe_train_test.prototxt]: Python API should not need to change this setting

File: data/lenet_dataframe_train_test.prototxt [hunk 3]
```diff
@@ -30,7 +30,7 @@ layer {
   }
   source_class: "com.yahoo.ml.caffe.ImageDataFrame"
   memory_data_param {
-    source: "hdfs:///projects/machine_learning/image_dataset/mnist_test_dataframe/"
+    source: "/Users/mridul/bigml/mnist_test_dataframe/"
```
Comment (anfeng) [inline data/lenet_dataframe_train_test.prototxt]: Python API should not need to change this setting

File: python_doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
+.. CaffeOnSparkPythonApi documentation master file, created by
+   sphinx-quickstart on Thu Mar 24 18:17:52 2016.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to CaffeOnSparkPythonApi's documentation!
```
Comment (anfeng) [inline python_doc/index.rst]: Welcome to CaffeOnSpark Python API

**Comment by mriduljain** [comment]
Completed all the above changes. Will checkin after exploring auto image generation shortly.

**Comment by anfeng** [comment]
I cannot view DLDemo.ipynb. Is the file corrupted?

**Comment by anfeng** [comment]
+1

**Comment by mriduljain** [comment]
Try downloading raw

**Pull request merged**